core/pkg/monitor/dcgm_exporter.go

```diff
@@ -0,0 +1,343 @@
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/wandb/wandb/core/internal/clients"
+	"github.com/wandb/wandb/core/internal/observability"
+	spb "github.com/wandb/wandb/core/pkg/service_go_proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+)
+
+type DCGMExporterParams struct {
+	// Full URL including query parameters.
+	URL string
+	// HTTP headers to include in the requests.
+	Headers map[string]string
+	// Logger to use for logging.
+	Logger *observability.CoreLogger
+	// Client is the base retryable HTTP client to use for requests.
+	Client *retryablehttp.Client
+}
+
+type DCGMExporter struct {
+	// url is the base URL of the OpenMetrics API endpoint.
+	url     string
+	queries []string
+
+	// v1api is the Prometheus API client.
+	v1api v1.API
+
+	// logger is the logger to log logs. ;)
+	logger *observability.CoreLogger
+}
+
+// parsePromQLURL parses a Prometheus API URL to get the base URL and query parameters.
```

Comment:
```suggestion
// ParsePromQLURL parses a Prometheus API URL to get the base URL and query parameters.
```

Reply:
or do you want this method to be private?

Reply:
no, I want to test it.